extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetColumnResolver.java

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.parquet.table.location;
+
+import io.deephaven.annotations.BuildableStyle;
+import io.deephaven.engine.table.impl.locations.TableKey;
+import io.deephaven.parquet.table.ParquetInstructions;
+import org.apache.parquet.column.ColumnDescriptor;
+import org.apache.parquet.schema.MessageType;
+import org.immutables.value.Value;
+
+import java.util.Map;
+
+/**
+ * A mapping between Deephaven column names and Parquet {@link ColumnDescriptor column descriptors}.
+ *
+ * TODO: describe better
+ */
+@Value.Immutable
+@BuildableStyle
+public abstract class ParquetColumnResolver {
+
+    /**
+     * {@link ParquetInstructions.Builder#setColumnResolverFactory(Factory)}
+     */
+    public interface Factory {
+
+        /**
+         * TODO: description
+         *
+         * @param tableKey the table key
+         * @param tableLocationKey the Parquet TLK
```

Comment:
You proposed `FileMetaData`. I think that's too limiting. I think you need to be able to get at Iceberg schemas per data file, for example, and you can't necessarily get that from Parquet metadata.